processor/memorylimiterprocessor/memorylimiter.go

```diff
@@ -90,225 +34,74 @@ func newMemoryLimiter(set processor.CreateSettings, cfg *Config) (*memoryLimiter
 		return nil, err
 	}
 
-	ml := &memoryLimiter{
-		usageChecker:   *usageChecker,
-		memCheckWait:   cfg.CheckInterval,
-		ticker:         time.NewTicker(cfg.CheckInterval),
-		readMemStatsFn: runtime.ReadMemStats,
-		logger:         logger,
-		mustRefuse:     &atomic.Bool{},
-		obsrep:         obsrep,
+	p := &memoryLimiterProcessor{
+		memlimiter: ml,
+		obsrep:     obsrep,
 	}
 
-	return ml, nil
-}
-
-func getMemUsageChecker(cfg *Config, logger *zap.Logger) (*memUsageChecker, error) {
-	memAllocLimit := uint64(cfg.MemoryLimitMiB) * mibBytes
-	memSpikeLimit := uint64(cfg.MemorySpikeLimitMiB) * mibBytes
-	if cfg.MemoryLimitMiB != 0 {
-		return newFixedMemUsageChecker(memAllocLimit, memSpikeLimit), nil
-	}
-	totalMemory, err := getMemoryFn()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get total memory, use fixed memory settings (limit_mib): %w", err)
-	}
-	logger.Info("Using percentage memory limiter",
-		zap.Uint64("total_memory_mib", totalMemory/mibBytes),
-		zap.Uint32("limit_percentage", cfg.MemoryLimitPercentage),
-		zap.Uint32("spike_limit_percentage", cfg.MemorySpikePercentage))
-	return newPercentageMemUsageChecker(totalMemory, uint64(cfg.MemoryLimitPercentage),
-		uint64(cfg.MemorySpikePercentage)), nil
+	return p, nil
 }
 
-func (ml *memoryLimiter) start(_ context.Context, host component.Host) error {
-	extensions := host.GetExtensions()
-	for _, extension := range extensions {
-		if ext, ok := extension.(interface{ GetBallastSize() uint64 }); ok {
-			ml.ballastSize = ext.GetBallastSize()
-			break
-		}
-	}
-	ml.startMonitoring()
+func (p *memoryLimiterProcessor) start(_ context.Context, host component.Host) error {
+	p.memlimiter.SetBallastSizeFromExtension(host)
```

Comment:
Does it mean that the processor now only works with the extension enabled? This should not be the case. The processor must keep working as is

Reply:
it's as before, it will set ballasize if it find the extension and set the field. otherwise, it will run without it

Reply:
renamed the func to make it clear